flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointedInputGate.java

```diff
@@ -180,40 +141,15 @@ private int offsetChannelIndex(int channelIndex) {
 			return Optional.empty();
 		}
 
-		if (endOfInputGate) {
```

Comment:
`return Optional.empty();` can be merged in condition. also remove `throws Exception`

> private Optional<BufferOrEvent> handleEmptyBuffer()  {
> 		if (inputGate.isFinished()) {
> 			isFinished = true;
> 		}
> 		return Optional.empty();
> 	}